packages/commonwealth/client/scripts/views/components/Polls/ViewPollVotesDrawer/ViewPollVotesDrawer.tsx

```diff
@@ -0,0 +1,274 @@
+  pollOptionsSummary: PollOptionSummary[]; // Renamed prop
+  totalVoteWeightInPoll: string | number; // Overall total voting power cast in the poll.
+  isOpen: boolean;
+  setIsOpen: Dispatch<SetStateAction<boolean>>;
+  tokenDecimals?: number | null | undefined;
+  topicWeight?: TopicWeightedVoting | null | undefined; // For formatting vote weights
+  communityId: string; // Community context, potentially for fetching profiles or other details.
+  onDownloadCsv: () => void; // Callback to trigger CSV download.
+  isLoading?: boolean; // To show loading state for votes
+};
+
+const getColumns = (
+  tokenDecimals?: number | null,
+  topicWeight?: TopicWeightedVoting | null,
+): CWTableColumnInfo[] => [
+  {
+    key: 'voter',
+    header: 'Voter',
+    numeric: false,
+    sortable: true,
+  },
+  {
+    key: 'option',
+    header: 'Chosen Option',
+    numeric: false,
+    sortable: true,
+  },
+  {
+    key: 'voteWeight',
+    header: 'Vote Weight',
+    numeric: true,
+    sortable: true,
+    tokenDecimals,
+    weightedVoting: topicWeight,
+  },
+  {
+    key: 'timestamp',
+    header: 'Timestamp',
+    numeric: true,
+    sortable: true,
+    chronological: true,
+  },
+];
+
+export const ViewPollVotesDrawer = ({
+  header,
+  votes,
+  pollOptionsSummary, // Use renamed prop
+  totalVoteWeightInPoll,
+  isOpen,
+  setIsOpen,
+  tokenDecimals,
+  topicWeight,
+  communityId,
+  onDownloadCsv,
+  isLoading = false,
+}: ViewPollVotesDrawerProps) => {
+  const columns = useMemo(
+    () => getColumns(tokenDecimals, topicWeight),
+    [tokenDecimals, topicWeight],
+  );
+
+  const tableState = useCWTableState({
+    columns,
+    initialSortColumn: 'timestamp',
+    initialSortDirection: APIOrderDirection.Desc, // Correctly pass APIOrderDirection.Desc
+  });
+
+  // Maps a VoterWithProfile object to the structure expected by CWTable's rowData.
+  const voterRow = (vote: VoterWithProfile) => {
+    // Find the display text for the chosen option.
+    // vote.option from ActualVoteAttributes is the ID/value of the chosen option.
+    const chosenOptionSummary = pollOptionsSummary.find(
+      // Use renamed prop
+      (opt) => opt.id === String(vote.option),
+    );
+
+    return {
+      voter: vote.profile?.name || vote.address,
+      option: chosenOptionSummary?.text || String(vote.option),
+      // Ensure calculated_voting_weight is a number for the table
+      voteWeight: vote.calculated_voting_weight
+        ? Number(vote.calculated_voting_weight)
+        : 0,
+      timestamp: vote.updated_at || vote.created_at, // These should be dates or date strings
+      _meta: {
+        address: vote.address,
+        avatarUrl: vote.profile?.avatarUrl,
+      },
+    };
+  };
+
+  const rowData = useMemo(() => {
+    if (isLoading || !votes) return [];
+    return votes.map(voterRow);
+  }, [votes, pollOptionsSummary, isLoading]); // Use renamed prop in dependency array
```

Comment:
React Hook useMemo has a missing dependency: 'voterRow'. Either include it or remove the dependency array